chromium/rules.js

```diff
@@ -209,7 +220,7 @@ RuleSets.prototype = {
       try {
         this.parseOneJsonRuleset(ruleset);
       } catch(e) {
-        log(WARN, 'Error processing ruleset:' + e);	
+        util.log(util.WARN, 'Error processing ruleset:' + e);
```

Comment:
Here is the WIP require stuff

Reply:
> I'm reluctant to add more functionality to this function because it is supposed to be a stub. Especially functionality that diverges from the functionality of the original function.

What about separate function? Something like `rules.reportError` which would throw in Node env and log otherwise.

Reply:
This doesn't need to be in the same PR though, I'm fine with a stub for now (please see comment below though).

Reply:
@RReverser I think it would be better for the function to be configurable by the user,  instead of having it switch behavior base on some global state. So something like `util.log.setBehavior(util.warn, ...)` would be a better API. Or, since this is specific to your needs, you could just monkey patch `console.warn` like:
```javascript
console.warn  = function() {
  throw(arguments);
}
```

Either way, I think adding this feature is out of scope of this PR.

Reply:
@cowlicks
`throw` is not a function, it is an operator:
```js
throw arguments;
```

Reply:
> instead of having it switch behavior base on some global state

Which is why I suggested having something like `rules.reportError` (or, now, `rules.settings.reportError` for consistency).

Reply:
Monkey-patching `console.warn` is worse since it affects all the Node.js libs then, not just `rules.js`. Point here is to restore native behaviour of errors (have them thrown and not silently swallowed) which doesn't work when using rules.js as a library because it catches all the errors.

Reply:
But yeah, it's out of scope either way.

Reply:
@RReverser oh sorry I think I misunderstood your API proposal. Having `rules.settings.reportError` be a setting for toggling error reporting would be reasonable. Agreed.